File: src/main/kotlin/com/github/fisherman08/micronautbooks/domain/book/Book.kt
package com.github.fisherman08.micronautbooks.domain.book

import com.github.fisherman08.micronautbooks.domain.writer.Writer

data class Book(
    val id: BookId,
    val title: BookTitle,
    val authors: Authors
) {
    companion object {
        fun register(title: BookTitle, authors: Collection<Writer>): Book =
            Book(
                id = BookId.generate(),
                title = title,
                authors = Authors.fromCollection(collection = authors)
            )
    }
}

File: webfrontend/src/domain/book/Book.ts
import { Writer } from "../writer/Writer";

const BookType = Symbol()

export class Book {
    [BookType]: any

    id: string
    title: string
    authors: Writer[]

    constructor(id: string, title: string, authors: Writer[]) {
        this.id = id
        this.title = title
        this.authors = authors
    }
}
